amanuensis/resources/message/__init__.py

```diff
@@ -1,138 +1,130 @@
-            logger.debug(f"send_message hubspot: {usernames}")
-            ec_users_results = fence_get_users(config=config, usernames=usernames)
-            logger.debug(f"fence_get_users, ec_users_results: {ec_users_results}")
-            ec_users = ec_users_results['users'] if 'users' in ec_users_results else None
-            # logger.debug(f"fence_get_users, ec_users: {ec_users}")
-           
-            if ec_users:
-                for ecu in ec_users:
-                    # logger.debug(f"send_message to: {ecu}")
-                    receivers.append(Receiver(receiver_id=ecu['id']))
-
-        #TODO get requestor email
-        # if logged_user_id is commettee memeber send to other commettee members and requestor
-        # otherwise send to committee memebers
-
-        return udm.send_message(session, logged_user_id, request_id, subject, body, receivers, usernames)
-
-
-def send_admin_message(project, consortiums, subject, body):
+    public_object_search_request = PublicObjectSearchRequest(
+        filter_groups=[
+            {
+                "filters": [
+                    {
+                        "value": committee,
+                        "propertyName": "disease_group_executive_committee", # TODO this might be committee membership and/or disease group affiliationin hubspot
+                        "operator": "EQ",
+                    }
+                ]
+            }
+        ],
+        properties=["email", "disease_group_executive_committee"],
+    )
+
+    try:
+        api_response = current_app.hubspot_client.crm.contacts.search_api.do_search(
+            public_object_search_request=public_object_search_request
+        )
+
+    except ApiException as e:
+        print("Exception when calling search_api->do_search: %s\n" % e)
+
+    payload = [result.properties for result in api_response.results]
+
+    return [
+        {key: val for key, val in property.items() if key in desired_properties}
+        for property in payload
+    ]
+
+def send_message(session, logged_user_id, request_id, subject, body):  
+
+    # Get consortium and check that the request exists
+    request = get_requests(session, user_id=logged_user_id, id=request_id)
+    # logger.debug("Request: " + str(request))
+    consortium_code = request.consortium_data_contributor.code
+    # logger.debug(f"Consortium Code: {consortium_code}")
+    
     # The hubspot oAuth implementation is on the way, but not supported yet.
-    hapikey =  config['HUBSPOT']['API_KEY']
-
-    if hapikey == "" or hapikey == "DEV_KEY":
-        logger.info('missing Hubspot API Key, skipping sending emails update.')
+    hapikey =  config['HUBSPOT']['ACCESS_TOKEN']
+    hubspot = current_app.hubspot_client(hubspot_auth_token=hapikey)
+
+    # Get EC members emails
+    # returns [ email, disease_group_executive_committee ]
+    committee = f"{consortium_code} Executive Committee Member"
+    hubspot_response = hubspot.get_contacts_by_committee(committee=committee)
```

Comment:
get_contacts_by_committee seems to be defined as a local function on line 21 of this file and not as part of the hubsport class.